### PR TITLE
ENH: Adds datetime arithmetic with timedeltas

### PR DESCRIPTION
**Pull request by llllllllll**
I wanted to be able to add timedeltas to interactive symbols so I made it so.

```python
In [5]: ds.quandl.vix.knowledge_date
Out[5]: 
   knowledge_date
0      1990-01-02
1      1990-01-03
2      1990-01-04
3      1990-01-05
4      1990-01-08
5      1990-01-09
6      1990-01-10
7      1990-01-11
8      1990-01-12
9      1990-01-15
...

In [6]: ds.quandl.vix.knowledge_date + datetime.timedelta(days=1)
Out[6]: 
   knowledge_date
0      1990-01-03
1      1990-01-04
2      1990-01-05
3      1990-01-06
4      1990-01-09
5      1990-01-10
6      1990-01-11
7      1990-01-12
8      1990-01-13
9      1990-01-16
...
```

datashape already supported this I just needed to add a `_(r)?add` and `_(r)?sub` to isdatelike

### CODE REVIEW

**Comment by mrocklin** [comment]
Whoa.  Cool.

**Comment by cpcloud** [comment]
noice

**Comment by cpcloud** [comment]
ah looks like a `dask` failure

**Comment by cpcloud** [comment]
@llllllllll doesn't seem related to your change

**Comment by cpcloud** [comment]
@llllllllll what backend are you using there?

**Comment by llllllllll** [comment]
blaze client serving a dictionary of alchemy metadata objects.

**Comment by cpcloud** [comment]
@llllllllll pull down master and push again, i just fixed the dask failure. it should take care of the failures we're seeing on travis here

**Comment by cpcloud** [comment]
Can you add a test for this? You can write it against postgres if you want. I have an instance running locally that I can test against and Travis-CI does as well.

**Comment by llllllllll** [comment]
I could do pg and pandas, I imagine this will be common on dataframes, series, and datetimeindidcies.

What is the connection string to use for pg?

**Comment by cpcloud** [comment]
`'postgresql://postgres@localhost/test::table'`

`table` can be whatever you want

**Comment by cpcloud** [comment]
gimme a sec and i'll add a pg db to travis

**Comment by llllllllll** [comment]
I imagine I will need to manage the creation of the table and data inside the test?

**Comment by cpcloud** [comment]
yeah, but pytest fixtures make that easy enough. if you have any questions about that i'm happy to help. odo also has an example here: https://github.com/ContinuumIO/odo/blob/master/odo/backends/tests/test_postgres.py#L35-L48

**Comment by llllllllll** [comment]
@cpcloud I rebased against  #1116 so your commits are pulled in here (under your name)

**Comment by cpcloud** [comment]
Excellent. I'll close the other PR then

File: blaze/compute/tests/test_sql_compute.py [hunk 1]
```diff
@@ -1508,3 +1538,29 @@ def test_not():
...
+        try:
+            yield t
+        finally:
+            drop(t)
+
+
+def test_timedelta_arith(sql_with_dts):
+    delta = timedelta(days=1)
+    dates = pd.Series(pd.date_range('2014-01-01', '2014-02-01'))
+    sym = symbol('s', '32 * datetime')
```
Comment (cpcloud) [inline blaze/compute/tests/test_sql_compute.py]: you should be able to do `symbol('s', discover(dates))` instead of explicitly writing down the dshape

**Comment by cpcloud** [comment]
@llllllllll tiny comment, which isn't critical to address, was just a thought. i hand tested this on redshift and works nicely. merging at some point today.

**Comment by cpcloud** [comment]
postgres looks like it has a lot of operations available for date arithmetic http://www.postgresql.org/docs/9.4/static/functions-datetime.html

might be nice to eventually implement some of these :)

**Comment by llllllllll** [comment]
I also added a test for numpy; however it only supports the np.timedelta64 objects instead of datetime.timedelta

**Comment by cpcloud** [comment]
PR on the way to skip numba optimizing when given a datetime / timedelta expression

**Comment by llllllllll** [comment]
Why, does numba not know how to handle these operations?

**Comment by cpcloud** [comment]
https://github.com/quantopian/blaze/pull/2

**Comment by cpcloud** [comment]
@llllllllll yeah it's not fully implemented in numba yet see here https://github.com/numba/numba/issues/1202

**Comment by llllllllll** [comment]
Is there a reason that the pr is against the quantopian fork, not the upstream?

**Comment by cpcloud** [comment]
yeah, because i wanted to use your test

**Comment by llllllllll** [comment]
kk

**Comment by llllllllll** [comment]
Rebased and fixed merge conflict

**Comment by llllllllll** [comment]
Looks like numba failed in the numpy test.

**Comment by cpcloud** [comment]
it looks like you might've rebased away my commits that fixed that issue

**Comment by llllllllll** [comment]
Were they on this branch or on master? I now remember that you had a fix for this

**Comment by cpcloud** [comment]
i submitted a PR to your fork

**Comment by cpcloud** [comment]
let me see if i have the branch lying around

**Comment by cpcloud** [comment]
it'd be in your reflog if you haven't run `git gc` since the rebase as well

**Comment by llllllllll** [comment]
I just cherry-picked them over

**Comment by cpcloud** [comment]
cool

**Comment by llllllllll** [comment]
sorry about that

**Comment by cpcloud** [comment]
no problem!

**Comment by cpcloud** [comment]
@llllllllll ok to merge on passing?

**Comment by llllllllll** [comment]
yup

**Pull request merged**